tests/conftest.py

```diff
@@ -269,72 +207,95 @@ def update_metadata_callback(req):
-# load any associated metadata or classification results
-API_DATA[
-    "GET::api/v1/samples\\?.*where=%7B%22project%22%3A\\+%224b53797444f846c4%22%7D.*"
-] = json.load(open(os.path.join("tests", "data", "api", "where_224b53797444f846c4.json")))
-
-for filename in os.listdir("tests/api_data"):
-    if not filename.endswith(".json"):
-        continue
-
-    resource = json.load(open(os.path.join("tests/api_data", filename)))
-    if filename.startswith("schema"):
-        continue  # Parse separately below
-
-    resource_name = filename.replace(".json", "")
-    resource_uri = "GET::api/v1/{}".format(resource_name)
-    API_DATA[resource_uri] = resource
-
-    # Then iterate through all instances
-    if isinstance(resource, list):
-        for instance in resource:
-            instance_uri = "GET::{}".format(instance["$uri"].lstrip("/"))
-            API_DATA[instance_uri] = instance
-
 SCHEMA_ROUTES = {}
-
-for filename in os.listdir("tests/api_data"):
-    if not filename.startswith("schema"):
-        continue
-
-    resource = json.load(open(os.path.join("tests/api_data", filename)))
-    if filename == "schema.json":
-        resource_uri = "GET::api/v1/schema$"
-    elif filename == "schema_all.json":
-        resource_uri = "GET::api/v1/schema\\?expand=all"
-    else:
-        resource_name = filename.replace(".json", "").split("_")[1]
-        resource_uri = "GET::api/v1/{}/schema".format(resource_name)
-
-    SCHEMA_ROUTES[resource_uri] = resource
+API_DATA_DIR = os.path.join("tests", "data", "api")
+
+for api_version in os.listdir(API_DATA_DIR):
+    api_root = os.path.join(API_DATA_DIR, api_version)
+
+    for resource_name in os.listdir(api_root):
+        resource_root = os.path.join(api_root, resource_name)
+
+        if resource_name == "schema":
+            for filename in os.listdir(resource_root):
+                if not filename.endswith(".json"):
+                    continue
+
+                if filename == "index.json":
+                    resource_uri = f"GET::api/{api_version}/schema$"
+                elif filename == "index_all.json":
+                    resource_uri = f"GET::api/{api_version}/schema\\?expand=all"
+                else:
+                    resource_name = filename.replace(".json", "")
+                    resource_uri = f"GET::api/{api_version}/{resource_name}/schema"
+
+                SCHEMA_ROUTES[resource_uri] = json.load(open(os.path.join(resource_root, filename)))
+        else:
+            for dirpath, _, filenames in os.walk(resource_root):
+                for filename in filenames:
+                    if filename not in {"index.json", "index.json.gz"}:
+                        continue
+
+                    filepath = os.path.join(dirpath, filename)
+                    if filepath.endswith(".json.gz"):
+                        # converting bytes (returned by gzip) to string is necessary in python 3.4
```

Comment:
Do we support python3.4? It's not even tested in the CI

Reply:
👍 removed, we support python 3.7+